gateway/src/main/java/com/mx/path/gateway/configuration/Configurator.java

```diff
@@ -257,6 +256,22 @@ private void populateFacilities(String clientId, ObjectMap map) {
         }
       });
     });
+
+    ensureDefaultFacilities(clientId);
   }
 
+  private void ensureDefaultFacilities(String clientId) {
+    //Default Event Bus
+    if (Facilities.getEventBus(clientId) == null) {
+      Facilities.addEventBus(
+          clientId,
+          gatewayObjectConfigurator.buildFromClass(
+              GatewayEventBus.class,
+              null,
+              clientId,
+              EventBus.class));
+    }
+
+    //Future defaults added here **
+  }
```

Comment:
Do we need to do use the object configurator here? Since it is defaults, there won't be any configuration binding needed. Can we just do something like `Facilities.addEventBus(clientId, new GatewayEventBus());`?

Reply:
I've simplified the code. Thoughts?

Reply:
👍  This looks better 👍 . It may be a good idea to pass in an empty `ObjectMap` instead of null, to future-proof a little.

```
gatewayObjectConfigurator.buildFromClass(
              GatewayEventBus.class,
              new ObjectMap(),
              clientId,
              EventBus.class));
```